### PR TITLE
[pytorch][build] Update PT 1.5.1 to Python 3.6.13

### PR DESCRIPTION
**Pull request by saimidu**
*Issue #, if available:*

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [x] (If applicable) I've documented below the tests I've run on the DLC image
- [X] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [x] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [x] I've verified the code change on the config file mentioned above has already been reverted

*Description:*
- Update PT 1.5.1 DLCs to use Python 3.6.13
- Upgrade cryptography version to `>=3.3.2`
- Upgrade pyyaml version to `>=5.4,<5.5`

*Tests run:*

*DLC image/dockerfile:*

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/dlc_tests/sanity/test_torchvision.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 
 @pytest.mark.model("N/A")
-def test_torchvision_nms_training(pytorch_training):
+def test_torchvision_nms_training(pytorch_training, pt16_and_above_only):
```
Comment (tejaschumbalkar) [inline test/dlc_tests/sanity/test_torchvision.py:15]: Why is this needed? Is the nms feature not available on the currently available PTv1.5.1 image?

Comment (saimidu) [reply test/dlc_tests/sanity/test_torchvision.py:15]: The feature being tested on this test seems to not have been added at the time.

**Review by tejaschumbalkar — COMMENTED** [review]
LGTM. Lets revert the build configs and safety/dependency test changes.

**Review by tejaschumbalkar — APPROVED** [review]
